components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAAdminService.java

```
@@ -0,0 +1,288 @@
+        } catch (CAException e) {
+            log.error("Error signing the CSR with serial no:" + serialNo + " , for " + validity + " days", e);
+            throw new CAException("Certificate could not be signed");
+        }
+    }
+
+    /**
+     * Lists all the certificate issued by the tenant CA.
+     *
+     * @return List of all tenant CA issued certificates
+     * @throws CAException
+     */
+    public Certificate[] listCertificates() throws CAException {
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            List<Certificate> certificateInfoList = certificateService.listCertificates(tenantDomain);
+            return certificateInfoList.toArray(new Certificate[certificateInfoList.size()]);
+        } catch (CAException e) {
+            log.error("Error listing certificates for tenant:" + tenantDomain, e);
+            throw new CAException("Error listing the certificates");
+        }
+    }
+
+    /**
+     * Lists all the certificates issued by the tenant CA filtered by the given status.
+     *
+     * @param status The status filter
+     * @return List of certificates with given status issued by tenant CA
+     * @throws CAException
+     * @see org.wso2.carbon.identity.certificateauthority.common.CertificateStatus
+     */
+    public Certificate[] listCertificatesWithStatus(String status) throws CAException {
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            List<Certificate> certificateInfoList = certificateService.listCertificates(status, tenantDomain);
+            return certificateInfoList.toArray(new Certificate[certificateInfoList.size()]);
+        } catch (CAException e) {
+            log.error("Error listing certificates with status " + status + " for tenant:" + tenantDomain, e);
+            throw new CAException("Error listing \"" + status + "\" certificates");
```

The original exception is swallowed here. Please fix.